test/ctl-e2e-test.sh

```diff
@@ -0,0 +1,183 @@
+          --lode-dir $UNIQUE_TEMP_DIR/lode \
+          --lode-password $LODE_PASSWORD \
+          $* --chrome-exe $(find_browser)" || rm -Rf $UNIQUE_TEMP_DIR
+}
+
+extract_settings() {
+    if [ -n "$1" ] && [ -f "$1" ]
+    then
+        tar xzf "$1"
+    fi
+}
+
+extract_crx() {
+    if [ -n "$1" ] && [ -f "$1" ]
+    then
+        unzip $1 -d $2 > /dev/zero
+    fi
+}
+
+
+run_browser () {
+    UNIQUE_TEMP_DIR=$(unique_temp_dir)
+    mkdir -p $UNIQUE_TEMP_DIR
+
+    extract_settings "$FLINT_SETTINGS"
+    extract_settings "$GERO_SETTINGS"
+    extract_settings "$LODE_SETTINGS"
+    extract_settings "$NAMI_SETTINGS"
+    extract_crx "$FLINT_CRX" "$UNIQUE_TEMP_DIR/flint"
+    extract_crx "$GERO_CRX" "$UNIQUE_TEMP_DIR/gero"
+    extract_crx "$LODE_CRX" "$UNIQUE_TEMP_DIR/lode"
+    extract_crx "$NAMI_CRX" "$UNIQUE_TEMP_DIR/nami"
+
+    $(find_browser) \
+        --load-extension=$UNIQUE_TEMP_DIR/flint,$UNIQUE_TEMP_DIR/gero,$UNIQUE_TEMP_DIR/nami,$UNIQUE_TEMP_DIR/lode \
+        --user-data-dir=$CHROME_PROFILE || rm -Rf $UNIQUE_TEMP_DIR
+
+}
+
+extract_settings_nami() {
+    extid=$1
+    tgt=$2
+
+    tar czf $tgt $CHROME_PROFILE/Default/Local\ Extension\ Settings/$extid/*
+}
+
+extract_settings_gero_flint() {
+    extid=$1
+    tgt=$2
+
+    tar czf $tgt \
+        $CHROME_PROFILE/Default/IndexedDB/chrome-extension_${extid}_0.indexeddb.leveldb \
+        $CHROME_PROFILE/Default/Extension\ State
+}
+
+extract_settings_lode() {
+    extid=$1
+    tgt=$2
+
+    tar czf $tgt $CHROME_PROFILE/Default/Local\ Extension\ Settings/$extid/* \
+                 $CHROME_PROFILE/Default/Extension\ State
+}
+
+MODE=""
+
+
+usage() {
+    echo "Usage: $0 cmd [options]"
+    echo " commands:"
+    echo "   run                  Run the tests"
+    echo "   browser              Start the browser in order to configure wallets"
+    echo "   settings [nami|gero|flint|lode] [-o out] "
+    echo "                        Dump wallet setting to out"
+    exit 1
+}
+
+cmd_run() {
+    run_tests
+}
+
+cmd_browser() {
+    run_browser
+}
+
+cmd_settings() {
+    case "$1" in
+        "flint")
+            CMD=extract_settings_gero_flint
+            EXTID=$EXTID_FLINT
+            TARGET=$FLINT_SETTINGS
+            ;;
+        "gero")
+            CMD=extract_settings_gero_flint
+            EXTID=$EXTID_GERO
+            TARGET=$GERO_SETTINGS
+            ;;
+        "nami")
+            CMD=extract_settings_nami
+            EXTID=$EXTID_NAMI
+            TARGET=$NAMI_SETTINGS
+            ;;
+        "lode")
+            CMD=extract_settings_lode
+            EXTID=$EXTID_LODE
+            TARGET=$LODE_SETTINGS
+            ;;
+    esac
+
+    shift 1
+    while getopts "o:" arg; do
+        case $arg in
+            o)
+                TARGET="$OPTARG"
+                ;;
+        esac
+    done
+
+    $CMD $EXTID $TARGET
+}
+
+if [ -z "$1" ]
+then
+    usage
+fi
+
+case "$1" in
+    "run")
+        shift 1
+        cmd_run $*
```

Comment:
Also applies to other cases below this one

```suggestion
        cmd_run "$@"
```

I don't see any reason to use `$*`